test/end-to-end/company.js

```diff
@@ -227,6 +228,290 @@ describe("Company endpoint", () => {
+            const adminReason = "yes.";
+
+            beforeAll(async () => {
+                await Company.deleteMany({});
+
+                [test_company, test_company_2, test_company_3, test_company_4, hidden_offer_company,
+                    disabled_offer_company, disabled_company, blocked_company] =
+                await Company.create(
+                    [test_company_data, test_company_data, test_company_data, test_company_data, test_company_data,
+                        test_company_data, disabled_company_data, blocked_company_data]);
+
+                await Account.deleteMany({});
+
+                await Account.create({
+                    email: test_user_admin.email,
+                    password: await hash(test_user_admin.password),
+                    isAdmin: true
+                });
+
+                await Account.create({
+                    email: test_user.email,
+                    password: await hash(test_user.password),
+                    company: test_company_4._id
+                });
+            });
+
+            afterEach(async () => {
+                await test_agent
+                    .delete("/auth/login")
+                    .expect(HTTPStatus.OK);
+            });
+
+            test("should fail if not a valid id", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const res = await test_agent
+                    .get("/company/123/profile")
+                    .send({ adminReason })
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                expect(res.body).toHaveProperty("error_code", ErrorTypes.VALIDATION_ERROR);
+                expect(res.body).toHaveProperty("errors");
+                expect(res.body.errors[0]).toHaveProperty("param", "companyId");
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.OBJECT_ID);
+            });
+
+            test("should fail if company does not exist", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const id = "111111111111111111111111";
+                const res = await test_agent
+                    .get(`/company/${id}/profile`)
+                    .send({ adminReason })
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                expect(res.body).toHaveProperty("error_code", ErrorTypes.VALIDATION_ERROR);
+                expect(res.body).toHaveProperty("errors");
+                expect(res.body.errors[0]).toHaveProperty("param", "companyId");
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.COMPANY_NOT_FOUND(id));
+            });
+
+            test("should succeed if the company has no offers", async () => {
```

Comment:
This test description may be misleading: one might think that having no offers => success